packages/@cdktf/provider-generator/lib/get/generator/emitter/struct-emitter.ts

```diff
@@ -282,7 +282,16 @@ export class StructEmitter {
       this.emitComplexListClass(struct);
     } else if (struct.nestingMode === "map") {
       this.emitComplexMapClass(struct);
+    } else if (struct.nestingMode === "maplist") {
+      this.emitComplexMapListClasses(struct);
+    } else if (struct.nestingMode === "mapset") {
+      this.emitComplexMapListClasses(struct);
+    } else if (struct.nestingMode === "listmap") {
+      this.emitComplexListMapClasses(struct, false);
+    } else if (struct.nestingMode === "setmap") {
+      this.emitComplexListMapClasses(struct, true);
     }
+    // TODO think if there's a good way to generalize these
```

Comment:
I think making them explicit (as done above) is okay – We'd like to rewrite the generator at some point anyway😄